src/main/scala/uk/sky/fs2/kafka/topicloader/TopicLoader.scala

```diff
@@ -34,56 +34,67 @@ trait TopicLoader {
-            _          <- OptionT.liftF(
-                            consumer.assign(logOffsets.keys) *>
-                              logOffsets.toNel.traverse { case (tp, o) => consumer.seek(tp, o.lowest) }
-                          )
-          } yield load(consumer, logOffsets)
-        }.getOrElse(Stream.empty)
-      }
-      .flatten
-
+      .flatMap(load(topics, strategy, _))
+
+  /** Stream that loads the specified topics from the beginning. When the latest current offsets are reached, the
+    * `onLoad` callback is evaluated, and the stream continues.
+    *
+    * @param topics
+    *   topics to load
+    * @param consumerSettings
+    *   [[fs2.kafka.ConsumerSettings]] for the given topics
+    * @param onLoad
+    *   A callback that will be evaluated on once the current offsets are reached
+    */
   def loadAndRun[F[_] : Async : LoggerFactory, K, V](
       topics: NonEmptyList[String],
       consumerSettings: ConsumerSettings[F, K, V]
   )(onLoad: Resource.ExitCase => F[Unit]): Stream[F, ConsumerRecord[K, V]] =
     KafkaConsumer
       .stream(consumerSettings)
-      .evalMap { consumer =>
-        {
-          for {
-            logOffsets    <- OptionT(logOffsetsForTopics(topics, LoadAll, consumer))
-            _             <- OptionT.liftF(
-                               consumer.assign(logOffsets.keys) *>
-                                 logOffsets.toNel.traverse { case (tp, o) => consumer.seek(tp, o.lowest) }
-                             )
-            preLoadStream <- OptionT.pure(load(consumer, logOffsets))
-          } yield preLoadStream.onFinalizeCase(onLoad) ++ consumer.records.map(_.record)
-        }.getOrElse(Stream.empty)
+      .flatMap { consumer =>
+        load(topics, LoadAll, consumer).onFinalizeCase(onLoad) ++ consumer.records.map(_.record)
       }
-      .flatten
 
   private def load[F[_] : Async : LoggerFactory, K, V](
-      consumer: KafkaConsumer[F, K, V],
-      logOffsets: NonEmptyMap[TopicPartition, LogOffsets]
-  ): Stream[F, ConsumerRecord[K, V]] = consumer.records.map(_.record).through(filterBelowHighestOffset(logOffsets))
+      topics: NonEmptyList[String],
+      strategy: LoadTopicStrategy,
+      consumer: KafkaConsumer[F, K, V]
+  ): Stream[F, ConsumerRecord[K, V]] = Stream
+    .eval({
+      for {
+        logOffsets <- OptionT(logOffsetsForTopics(topics, strategy, consumer))
+        _          <- OptionT.liftF(consumer.assign(logOffsets.keys))
+        _          <- OptionT.liftF(logOffsets.toNel.traverse((tp, o) => consumer.seek(tp, o.lowest)))
+      } yield consumer.records.map(_.record).through(filterBelowHighestOffset(logOffsets))
+    }.getOrElse(Stream.empty))
+    .flatten
 
   private def filterBelowHighestOffset[F[_] : Monad : LoggerFactory, K, V](
       logOffsets: NonEmptyMap[TopicPartition, LogOffsets]
   ): Pipe[F, ConsumerRecord[K, V], ConsumerRecord[K, V]] = {
+    val nonEmptyOffsets =
+      logOffsets.toSortedMap.filter((_, o) => o.highest > 0)
```

Comment:
This is the meat of the fix, previously we were filtering off non-empty offsets before this point. This is important as it meant we were not subscribing to empty partitions. for loadAndRun we want to subscribe to all partitions, but only `load` from those that are non-empty.